cli/tools/test.rs

```diff
@@ -76,12 +76,39 @@ pub enum TestOutput {
 
 #[derive(Debug, Clone, PartialEq, Deserialize)]
 #[serde(rename_all = "camelCase")]
-pub enum TestResult {
+pub struct TestResult {
```

Comment:
Revert this type to the enum it's supposed to be

Reply:
Could you clarify why?

Reply:
An Rust enum can represent N-state very well, extracting it into a struct you end up duplicating things. The same result can be used seamlessly for both test result and step results. Here you end up duplicating errors in both the test step and test step result. They're just test results, not a specific kind of test result.